src/fs.cpp

```diff
@@ -249,9 +249,9 @@ void ofstream::close()
 #else // __GLIBCXX__
 
 #if BOOST_VERSION >= 107700
-static_assert(sizeof(*BOOST_FILESYSTEM_C_STR(fs::path())) == sizeof(wchar_t),
+static_assert(sizeof(*BOOST_FILESYSTEM_C_STR(boost::filesystem::path())) == sizeof(wchar_t),
 #else
-static_assert(sizeof(*fs::path().BOOST_FILESYSTEM_C_STR) == sizeof(wchar_t),
+static_assert(sizeof(*boost::filesystem::path().BOOST_FILESYSTEM_C_STR) == sizeof(wchar_t),
```

Comment:
Can probably remove all of this code, because `BOOST_FILESYSTEM_C_STR` is no longer used? If yes, maybe even split up this change

Reply:
I'm happy to base #20744 on this change, if this is the direction we are taking, and save splitting the Boost removal over several PRs.

Reply:
If we want to move forward with removing `boost::fs`, it could even make sense to split up the build system changes, so that other pull requests can start using the other C++17 features.

Reply:
> I'm happy to base #20744 on this change, if this is the direction we are taking, and save splitting the Boost removal over several PRs.

Yeah, sorry, I've been a slow iterating over the documentation and variations on the idea, but I'll post an updated #20744 and rebased version of this PR today